### PR TITLE
Update comment count to not include dialogue responses and make new-dialogue notification clickable

### PR DESCRIPTION
**Pull request by Discordius**
This makes it so the commentCount doesn't include dialogue responses anymore. Also makes the dialogue counter clickable so that it expands the PostsItem to show new dialogue responses and comments.

Requires running the filterDebateResponses migration.

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1201302964208280/1205454856116556) by [Unito](https://www.unito.io)

### CODE REVIEW

File: packages/lesswrong/server/manualMigrations/2023-09-08-filterDebateResponses.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import { registerMigration } from './migrationUtils';
```
Comment (jpaddison3) [inline packages/lesswrong/server/manualMigrations/2023-09-08-filterDebateResponses.ts:1]: Old man Habryka too lazy to learn how to write automatic migrations

File: packages/lesswrong/server/callbacks/commentCallbacks.ts [hunk 2]
```diff
@@ -128,7 +129,7 @@ getCollectionHooks("Comments").removeAsync.add(async function CommentsRemovePost
   const { postId } = comment;
 
   if (postId) {
-    const postComments = await Comments.find({postId}, {sort: {postedAt: -1}}).fetch();
+    const postComments = await Comments.find({postId, debateResponse: false, deleted: false, deletedPublic: false}, {sort: {postedAt: -1}}).fetch();
```
Comment (jpaddison3) [inline packages/lesswrong/server/callbacks/commentCallbacks.ts:132]: I would guess you'd want

{deleted: false} OR {deleted: true, deletedPublic: false}

With the current version you can go ahead and get rid of deletedPublic, it's not doing anything extra.

I would also note that this is not null-safe, although my current view of the database shows that, while the column is nullable, none of the values are null.

File: packages/lesswrong/server/callbacks/commentCallbacks.ts [hunk 3]
```diff
@@ -91,6 +91,7 @@ getCollectionHooks("Comments").newValidate.add(async function createShortformPos
 });
 
 getCollectionHooks("Comments").newSync.add(async function CommentsNewOperations (comment: DbComment) {
+  if (comment.debateResponse) return comment
```
Comment (jpaddison3) [inline packages/lesswrong/server/callbacks/commentCallbacks.ts:94]: Don't you still want the author's read statuses to get updated?

File: packages/lesswrong/components/posts/LWPostsItem.tsx [hunk 4]
```diff
@@ -466,8 +472,8 @@ const LWPostsItem = ({classes, ...props}: PostsList2Props) => {
               <PostsUserAndCoauthors post={post} abbreviateIfLong={true} newPromotedComments={hasNewPromotedComments} tooltipPlacement="top"/>
             </PostsItem2MetaInfo>}
 
-            {!!post.unreadDebateResponseCount && <PostsItem2MetaInfo>
-              <div className={classes.unreadDebateResponseCount}>
+            {!!post.unreadDebateResponseCount && <PostsItem2MetaInfo className={classes.unreadDebateResponseContainer}>
+              <div className={classes.unreadDebateResponseCount} onClick={toggleComments}>
```
Comment (jpaddison3) [inline packages/lesswrong/components/posts/LWPostsItem.tsx:476]: I haven't checked if this is sensible

**Review by jpaddison3 — APPROVED** [review]
ASTLAC

File: packages/lesswrong/server/manualMigrations/2023-09-08-filterDebateResponses.ts (hunk 1 above)
Comment (Discordius) [reply packages/lesswrong/server/manualMigrations/2023-09-08-filterDebateResponses.ts:1]: lol, I would actually like to learn about how to do this automatically. Will figure it out next week.

**Comment by jpaddison3** [comment]
@Raemon c'mon. Please look at the comments before you merge every approved PR.

**Pull request merged**